internal/controller/runtime/fsm/runtime_fsm_configure_auditlog.go

```diff
@@ -0,0 +1,36 @@
+package fsm
+
+import (
+	"context"
+
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func sFnConfigureAuditLog(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	m.log.Info("Configure Audit Log state")
+
+	wasAuditLogEnabled, err := m.AuditLogging.Enable(ctx, s.shoot)
+	if err != nil && !wasAuditLogEnabled {
```

Comment:
I think we can replace `if err != nil && !wasAuditLogEnabled`  with `if err != nil` .